### ISSUE TITLE
Poor error message from unsupported RoleAnnotations

### ISSUE THREAD

**Issue by treeowl** [augustss/MicroHs#73]
```haskell
{-# LANGUAGE RoleAnnotations #-}
module R where

data R r = R
type role R nominal
```

MicroHs tells me

```
mhs: error: "Depth.hs": line 5, col 11:
  found:    R
  expected: SymOper
```

It looks like your parser sees that `role` is lower case, and therefore expects the next token to be an operator. `R`, of course, isn't one. If you want to maximize GHC compatibility, the best thing is likely to parse type role annotations and then discard them. If you want to come up with your own safe coercion story in the future, though, it might be better to recognize when `type role` is followed by a capitalized name or an operator in parentheses and to point out in the error message that MicroHs doesn't support role annotations.

**Comment by augustss** [comment]
Yeah, I can add parsing of role annotations.

**Comment by augustss** [comment]
Fixed in 670601abe133212d98bb41f8a187b361229ef9a4
